demos/inpage/src/helpers/calculateUserOpHash.ts

```diff
@@ -0,0 +1,49 @@
+import { ethers } from 'ethers';
+
+import EthereumRpc from '../EthereumRpc';
+
+export default function calculateUserOpHash(
+  userOp: EthereumRpc.UserOperation,
+  entryPoint: string,
+  chainId: number,
+) {
+  return ethers.keccak256(
+    ethers.AbiCoder.defaultAbiCoder().encode(
+      ['bytes32', 'address', 'uint256'],
+      [simpleHash(userOp), entryPoint, chainId],
+    ),
+  );
+}
+
+function simpleHash(userOp: EthereumRpc.UserOperation) {
+  return ethers.keccak256(pack(userOp));
+}
+
+function pack(userOp: EthereumRpc.UserOperation) {
```

Comment:
Consider renaming ("encodeUserOp"?) since "packed" has meaning in the encoding context.